docs/python/examples/stock-tracker.md

```diff
@@ -0,0 +1,108 @@
+---
+displayed_sidebar: examplesSidebar
+sidebar_position: 10
+title: Stock Tracker
+---
+
+In this example, we use DBOS to build and deploy an app that tracks stock prices and send SMS alerts when the price of a stock crosses a certain threshold. We certainly don't want to miss a good opportunity to trade, so it's important this workflow is punctual and reliable!
+
+You can see the application live [here](https://max-stock-prices.cloud.dbos.dev/). All source code is [available on GitHub](https://github.com/dbos-inc/dbos-demo-apps/tree/main/python/stock-prices).
+
+## Import and Initialize the App
+
+Let's start off by importing the necessary libraries and initializing the app.
+
+```python
+import os
+import datetime
+import threading
+import yfinance as yf
+from twilio.rest import Client
+from dbos import DBOS
+from schema import stock_prices, alerts
+
+DBOS()
+```
+
+## Stock Prices Watcher as an online cron job
+
+We'll use a [DBOS Scheduled Workflow](../tutorials/scheduled-workflows) to make this job punctual and reliable.
+Under the hood, DBOS leverages Postgres to store the job's execution status and schedule, ensuring that the job is executed even if the server is restarted.
+
+```python
+@DBOS.scheduled('* * * * *')
+@DBOS.workflow()
+def fetch_stock_prices_workflow(scheduled_time: datetime, actual_time: datetime):
+    # For simplicity, let's declare inline a list of stock symbols to track
+    symbols = ['AAPL', 'GOOGL', 'AMZN', 'MSFT', 'TSLA', 'NVDA']
+    # Fetch registered alerts
+    registered_alerts = fetch_alerts()
+    # Fetch stock prices for each symbol
+    for symbol in symbols:
+        price = fetch_stock_price(symbol)
+        save_to_db(symbol, price)
+        # If there is a registered alert for that symbol, send a SMS if the price is above the alert threshold
+        if registered_alerts and symbol in registered_alerts:
+            if price > registered_alerts[symbol].price_threshold:
+                send_sms_alert(symbol, price, registered_alerts[symbol].phone_number)
+```
+
+## Fetching stock prices
+
+We'll use the `yfinance` library to fetch stock prices and write a DBOS [step](../tutorials/step-tutorial) to make it durable.
+
+```python
+@DBOS.step()
+def fetch_stock_price(symbol):
+    stock = yf.Ticker(symbol)
+    data = stock.history(period="1d")
+    if data.empty:
+        raise ValueError("No stock data found for the symbol.")
+    DBOS.logger.info(f"Stock price for {symbol} is {data['Close'].iloc[0]}")
+    return data['Close'].iloc[0]
+```
+
+## Sending SMS alerts
+
+We'll use the Twilio API to send SMS alerts and write a DBOS [step](../tutorials/step-tutorial) to make it durable. You can sign up for a free Twilio account at https://www.twilio.com/try-twilio. We'll need a few environment variables to store our Twilio account SID, auth token and phone number.
+
+```python
+twilio_account_sid = os.environ.get('TWILIO_ACCOUNT_SID')
```

Comment:
Where do the alerts come from?

Reply:
The DBOS workflow will call the Twilio API. Twilio is responsible for delivering the SMS.